Automattic/wp-calypso
# Full Site Editing: Simplify how FSE blocks fetch posts

#### Changes proposed in this Pull Request

In #32529, the FSE blocks fetching posts logic was based upon an `apiFetch` fired from a custom made `fetchPost` async function.
While not wrong per se, it forced an API call regardless of the presence of such post in state, and on top of that, didn't take advantage of the Redux/`withSelect` perks.

* Remove the `fetchPost` utility, now unnecessary.
* Use `select( 'core' ).getEntityRecord( 'postType', postType, postId )` instead of the more convoluted custom mix of `getEntityRecord` and `apiFetch` proposed in `fetchPost`.
* Make full use of HOC composition of `withState` and `withSelect` to request a new post (via `getEntityRecord`) only when a post has been selected and its ID and type have been stored as attribute, or into state (in the Content block, that doesn't persist it).

#### Testing instructions

* Smoke test Content and Template blocks:
  - Select a post and make sure the preview is displayed as expected.
  - Flip between edit and preview mode.
  - Save and reload, and check that Template shows up previewing the selected post. Note: this is currently broken because of #32657; a fix has been proposed in #32717).
* Make sure the blocks are saved as expected:
  - Content shouldn't persist anything.
  - Template should save the selected post ID and type as block attributes.

Fixes Dennis' concerns in https://github.com/Automattic/wp-calypso/pull/32529#pullrequestreview-232399549 (well, at least tries to 😄)
